src/shared/Database/DatabaseSqlite.cpp

```diff
@@ -0,0 +1,392 @@
+/*
+ * This file is part of the CMaNGOS Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include "QueryResultSqlite.h"
+#include <cstdint>
+#include <cstring>
+#include <sqlite3.h>
+#ifdef DO_SQLITE
+
+#include "Util/Util.h"
+#include "Policies/Singleton.h"
+#include "Platform/Define.h"
+#include "Multithreading/Threading.h"
+#include "DatabaseEnv.h"
+#include "Util/Timer.h"
+#include "DatabaseSqlite.h"
+
+size_t DatabaseSqlite::db_count = 0;
+
+void DatabaseSqlite::ThreadStart()
+{ };
+
+void DatabaseSqlite::ThreadEnd()
+{ };
+
+DatabaseSqlite::DatabaseSqlite()
+{
+    // before first connection
+    if (db_count++ == 0);
+}
+
+DatabaseSqlite::~DatabaseSqlite()
+{
+    StopServer();
+    if (--db_count == 0);
+}
+
+SqlConnection* DatabaseSqlite::CreateConnection()
+{
+    return new SQLiteConnection(*this);
+}
+
+SQLiteConnection::~SQLiteConnection()
+{
+    FreePreparedStatements();
+    //mysql_close(mSqlite);
+    sqlite3_close(mSqlite);
+}
+
+bool SQLiteConnection::Initialize(const char* infoString)
+{
+    if (sqlite3_open(infoString, &mSqlite) != SQLITE_OK)
+    {
+        sLog.outError("Could not open SQLite database");
+        return false;
+    }
+    sqlite3_exec(mSqlite, "PRAGMA journal_mode=WAL;", 0, 0, 0);
+    sqlite3_busy_timeout(mSqlite, 2);
+
+    Tokens tokens = StrSplit(infoString, ";");
+
+    DETAIL_LOG("Connected to SQLite database at %s", infoString);
+
+
+    // set connection properties to UTF8 to properly handle locales for different
+    // server configs - core sends data in UTF8, so MySQL must expect UTF8 too
+//    Execute("SET NAMES `utf8`");
+//    Execute("SET CHARACTER SET `utf8`");
+
+    return true;
+}
+
+bool SQLiteConnection::_Query(const char* sql, sqlite3_stmt** pStmt)
+{
+    if (!mSqlite)
+        return false;
+
+    uint32 _s = WorldTimer::getMSTime();
+    int result;
+    result = sqlite3_prepare_v2(mSqlite, sql, -1, pStmt, NULL);
+    if (result != SQLITE_OK) {
+        sLog.outErrorDb("SQL: %s", sql);
+        sLog.outErrorDb("query ERROR: %s", sqlite3_errmsg(mSqlite));
+        return false;
+    }
+
+    return true;
+}
+
+QueryResult* SQLiteConnection::Query(const char* sql)
```

Comment:
This is one function that changed to return a ```std::unique_ptr``` so adjust this function accordingly (see similar changes in MySQL/PostgreSQL version).